### ISSUE TITLE
Make search bar show only tags that have search results

### ISSUE THREAD

**Issue by jjchen20** [razzius/hms-weave#105]

**Comment by jjchen20** [comment]
Additional ask: Change search bar text from "Search" to "Click or type something and press enter." I think people don't know that free text searching is an option.

**Comment by razzius** [comment]
Moved search bar text to new issue

**Comment by razzius** [comment]
![image](https://user-images.githubusercontent.com/2244895/75402127-ee180600-58b8-11ea-91d5-a449093cc983.png)

I've made some progress on this, and a question has come up. Should we require hospital affiliations to be made "public" or should they always be public?

**Comment by razzius** [comment]
On staging, now, the /browse page will only show tags that are public. So we'll have to do some administration but the code should work.

**Comment by jjchen20** [comment]
Hospital affiliations should always be "public"

**Comment by jjchen20** [comment]
Showing the available public tags requires first clicking away then clicking back to the search box. Would be great if the drop down list of available tags comes up by clicking in the box, without needing to click away and then back.

**Comment by jjchen20** [comment]
The fluidity of the searching with tags that have search results is dependent on #77

**Comment by razzius** [comment]
This feature is pretty much done and deployed to staging. Degrees are not loading dynamically yet since we have special cases of DMD / DDS and MD / DO which need additional information to specify the tags are "alias" tags or something.

**Comment by jjchen20** [comment]
For the hospitals, right now the only way to remove a selected hospital filter is to clear the entire search. There should be a way to remove the hospital filter option (i.e. with a small x next to the hospital name) without clearing the entire search, in case several tags and/or degree filtering options are selected.

**Comment by razzius** [comment]
Good point @jjchen20. I'm adding that to a new issue: https://github.com/razzius/hms-weave/issues/176
